src/m365/spp/commands/model/model-remove.spec.ts

```diff
@@ -0,0 +1,200 @@
+  });
+
+  it('is the confirmation prompt called with title information', async () => {
+    const confirmationStub = sinon.stub(cli, 'promptForConfirmation').resolves(false);
+
+    await command.action(logger, { options: { siteUrl: 'https://contoso.sharepoint.com/sites/portal', title: 'modelTitle' } });
+    assert(confirmationStub.calledOnce);
+    assert(confirmationStub.args[0][0].message.startsWith('Are you sure you want to remove model - modelTitle ?'));
+  });
+
+  it('deletes model by id', async () => {
+    const stubDelete = sinon.stub(request, 'delete').callsFake(async (opts) => {
+      if (opts.url === `https://contoso.sharepoint.com/sites/portal/_api/machinelearning/models/getbyuniqueid('9b1b1e42-794b-4c71-93ac-5ed92488b67f')`) {
+        return;
+      }
+
+      throw 'Invalid request';
+    });
+
+    await command.action(logger, { options: { siteUrl: 'https://contoso.sharepoint.com/sites/portal', id: '9b1b1e42-794b-4c71-93ac-5ed92488b67f', force: true } });
+    assert(stubDelete.calledOnce);
+  });
+
+  it('does not delete model when confirmation is not accepted', async () => {
+    sinon.stub(cli, 'promptForConfirmation').resolves(false);
+
+    const stubDelete = sinon.stub(request, 'delete');
+
+    await command.action(logger, { options: { siteUrl: 'https://contoso.sharepoint.com/sites/portal', id: '9b1b1e42-794b-4c71-93ac-5ed92488b67f' } });
+    assert(stubDelete.notCalled);
+  });
+
+  it('deletes model by id without force when confirmation is accepted', async () => {
+    sinon.stub(cli, 'promptForConfirmation').resolves(true);
+
+    const stubDelete = sinon.stub(request, 'delete').resolves();
+
+    await command.action(logger, { options: { siteUrl: 'https://contoso.sharepoint.com/sites/portal', id: '164720c8-35ee-4157-ba26-db6726264f9d' } });
+    assert(stubDelete.calledOnce);
+  });
+
+  it('deletes model when the the site URL has trailing slash', async () => {
+    const stubDelete = sinon.stub(request, 'delete').callsFake(async (opts) => {
+      if (opts.url === `https://contoso.sharepoint.com/sites/portal/_api/machinelearning/models/getbyuniqueid('9b1b1e42-794b-4c71-93ac-5ed92488b67f')`) {
+        return;
+      }
+
+      throw 'Invalid request';
+    });
+
+    await command.action(logger, { options: { siteUrl: 'https://contoso.sharepoint.com/sites/portal/', id: '9b1b1e42-794b-4c71-93ac-5ed92488b67f', force: true } });
+    assert(stubDelete.calledOnce);
+  });
+
+  it('deletes model by title', async () => {
+    const stubDelete = sinon.stub(request, 'delete').callsFake(async (opts) => {
+      if (opts.url === `https://contoso.sharepoint.com/sites/portal/_api/machinelearning/models/getbytitle('ModelName')`) {
+        return '';
+      }
+
+      throw 'Invalid request';
+    });
+
+    await command.action(logger, { options: { siteUrl: 'https://contoso.sharepoint.com/sites/portal', title: 'ModelName', force: true } });
+    assert(stubDelete.calledOnce);
+  });
+});
```

Comment:
In case this test fails, we want the delete request to return, otherwise we'll get a timeout instead of a failing assert.

```suggestion
    const stubDelete = sinon.stub(request, 'delete').resolves();
```